### ISSUE TITLE
False positives of Qt Linguist with Typescript files

### ISSUE THREAD

**Issue by frankdugan3** [AlDanial/cloc#324]
I'm getting a number of false positive for Qt Linguist on simple Typescript files. Looking back at 9afbdd88422872021bdf3bed26bc9cb8608a4416, I think any or all of the following tweaks would help eliminate the issue:
1. It's becoming more common for JS and TS to drop semicolons, and use `const` and `let` instead of `var`, so adding a few more keywords for points would help:
```
9221: ++$tscript_points  if /^\s*(var|const|let|class|document)\b/;
```
2. It's likely for the existing point system to score a 0 for both languages with simple Typescript files, but it currently defaults Qt Linguist, which I'm guessing is more rare of the two. I think it would make more sense to default to Typescript if the scores match, or alternatively if the score is zero. I don't know much about QT Linguist, but a quick glance at the file format for `.ts` looks like the score would always be higher than 0 if it is QT Linguist with the current scoring system.
```
9226: if ($tscript_points >= $linguist_points) {
```

**Comment by AlDanial** [comment]
Thanks for the tips; they've been merged in.  Your timing is good as I plan to make a release by the weekend.

It would be helpful to me if you could post a dummied-up TS file that, before this fix, would have been misidentified as Qt Linguist so that I can add it to my test suite.

**Comment by frankdugan3** [comment]
Happy to help! This is valid, working Typescript and a false positive for Qt Linguist.

```
import { createServer } from 'http'

export const server = createServer((req, res) => {
  res.writeHead(200, { 'Content-type': 'text/plain' })
  res.write('Hello world!')
  res.end()
}).listen(8080)
```

**Comment by AlDanial** [comment]
Update appears in the just-released v 1.78.
